### ISSUE TITLE
Deploy To Oryx Failed

### ISSUE THREAD

**Issue by 2xOu7** [microsoft/Oryx#2341]
### Bug Report

- At what date and time did you most recently experience the problem?
1/16/2024 at 4:53 PM PST

- Where did you experience the problem? E.g. Azure Web Apps, Azure Functions, Azure Container Registry, or offline use.
Azure Web Apps

- If your repo is publicly available please share its URL:

- What happened?

- What did you expect or want to happen?

I wanted my app to deploy

- How can we reproduce it?

- Do you have log output? Please include between the backticks:

4:50:59 PM resonance-app: Updating submodules.
4:51:01 PM resonance-app: Preparing deployment for commit id '8af96abc-b'.
4:51:01 PM resonance-app: PreDeployment: context.CleanOutputPath False
4:51:01 PM resonance-app: PreDeployment: context.OutputPath /home/site/wwwroot
4:51:01 PM resonance-app: Repository path is /tmp/zipdeploy/extracted
4:51:01 PM resonance-app: Running oryx build...
4:51:01 PM resonance-app: Command: oryx build /tmp/zipdeploy/extracted -o /home/site/wwwroot --platform nodejs --platform-version 20 -p virtualenv_name= --log-file /tmp/build-debug.log  -i /tmp/8dc16f6602b0893 -p compress_node_modules=tar-gz | tee /tmp/oryx-build.log
4:51:03 PM resonance-app: Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
4:51:03 PM resonance-app: You can report issues at https://github.com/Microsoft/Oryx/issues
4:51:03 PM resonance-app: Oryx Version: 0.2.20230829.1, Commit: 5bd3088141516cdbd815d62ada052654e91f0429, ReleaseTagName: 20230829.1
4:51:03 PM resonance-app: Build Operation ID: 5f611e40298d2ce2
4:51:03 PM resonance-app: Repository Commit : 8af96abc-b943-4156-a012-b119f25ab0b8
4:51:03 PM resonance-app: OS Type           : bookworm
4:51:03 PM resonance-app: Image Type        : githubactions
4:51:03 PM resonance-app: Detecting platforms...
4:51:12 PM resonance-app: Deployment failed.

**Comment by 2xOu7** [comment]
Other logs I see from Azure Portal (weirdly it is saying python is not supported): 

Error: Platform 'python' version '3.8' is unsupported. Supported versions: 
Error: Platform 'python' version '3.8' is unsupported. Supported versions:\n/bin/bash -c "oryx build /tmp/zipdeploy/extracted -o /home/site/wwwroot --platform nodejs --platform-version 20 -p virtualenv_name= --log-file /tmp/build-debug.log  -i /tmp/8dc16f84dfa9ee6 -p compress_node_modules=tar-gz | tee /tmp/oryx-build.log ; exit $PIPESTATUS "

**Comment by simonjj** [comment]
@2xOu7 based on the output it seems like the detection isn't picking up the right language. Would you mind sharing the root of your directory here please?

**Comment by diwic** [comment]
I'm having the same/similar issue - it complained that PHP 8.0 was not supported for my node.js app. I worked around it by changing from Nodejs 20 to Nodejs 18. Maybe Nodejs 20 support is missing in the deployed instance of Oryx?

**Comment by rishi5565** [comment]
> I'm having the same/similar issue - it complained that PHP 8.0 was not supported for my node.js app. I worked around it by changing from Nodejs 20 to Nodejs 18. Maybe Nodejs 20 support is missing in the deployed instance of Oryx?

Yeah changing this worked for me too. But I cant use Nodejs 20 now. Seems like a bug.

**Comment by dandmcd** [comment]
I'm having this issue as well as we are switching to Nodejs 20-lts from18..  First time I deployed it worked fine with NodeJS 20 in the pipeline yml file.  Deploying again for any other reason using Azure App Service pipeline results in `Error: Platform 'php' version '8.0' is unsupported. Supported versions: 8.1.22, 8.2.9, 8.1.22, 8.2.9` and the deploy fails.

**Comment by allanctan** [comment]
I'm getting same error on Node 20 
Error: Platform 'php' version '8.0' is unsupported. Supported versions: 8.1.22, 8.2.9, 8.1.22, 8.2.9
